packages/lds/src/index.ts

```diff
@@ -104,6 +108,10 @@ export default async function subscribeToLogicalDecoding(
   async function loop() {
     try {
       const rows = await client.getChanges(null, 500);
+      if (isClosing) {
+        // Skip processing this data and exit.
+        return;
+      }
```

Comment:
This is the main part I need to think about if it's acceptable or not - it's effectively throwing away LDS data.

Reply:
I wouldn't throw it away. I don't think there's anything wrong with firing a few other events, even if they don't reach anyone. Better than discarding data that would have been needed.

OTOH, teardown is only available in test environments, so it might not matter at all… Or is `ldSubscription.close()` also used for graceful exit in prod?

Reply:
Yeah, I'm not happy with it... but also calling stuff _after_ release resolves definitely feels wrong. I'm thinking we should make the release method await the loop exiting, but that's a bit more code than I can write via the GitHub interface :laughing: Fancy taking it on? I was thinking wrapping `loop` in a promise we can `await` might work. Or maybe using a deferred: https://github.com/graphile/worker/blob/main/src/deferred.ts

Reply:
Having a `let onClose: (() => void) | null` instead of the `isClosing` flag, with `return new Promise(resolve => { onClose = resolve; })` might work? I'm only using Github here either, that's why I failed the prettier check…

Reply:
Yeah one of us should probably check this out and actually write it in an editor. And... you know... see if it works :wink: